src/main/java/com/alvarium/annotators/PkiHttpAnnotator.java

```diff
@@ -0,0 +1,146 @@
+import com.alvarium.sign.SignProviderFactory;
+import com.alvarium.sign.SignatureInfo;
+import com.alvarium.sign.SignType;
+import com.alvarium.utils.Encoder;
+import com.alvarium.utils.PropertyBag;
+
+import org.apache.http.client.methods.HttpUriRequest;
+
+class PkiHttpAnnotator extends AbstractAnnotator implements Annotator {
+  private final HashType hash;
+  private SignatureInfo signature;
+  private final AnnotationType kind;
+
+  protected PkiHttpAnnotator(HashType hash, SignatureInfo signature) {
+    this.hash = hash;
+    this.signature = signature;
+    this.kind = AnnotationType.PKIHttp;
+  }
+
+  public Annotation execute(PropertyBag ctx, byte[] data) throws AnnotatorException {
+    final String key = super.deriveHash(hash, data);
+
+    String host;
+    try {
+      host = InetAddress.getLocalHost().getHostName();
+    } catch (UnknownHostException e) {
+      throw new AnnotatorException("Cannot get host name", e);
+    }
+
+    HttpUriRequest request = ctx.getProperty(AnnotationType.PKIHttp.name(), HttpUriRequest.class);
+    ParseResult parsed; 
+    try {
+      parsed = new ParseResult(request);
+    } catch (URISyntaxException e) {
+      throw new AnnotatorException("Invalid request URI", e);
+    } catch (ParseResultException e) {
+      throw new AnnotatorException("Error parsing the request", e);
+    }
+    final Signable signable = new Signable(parsed.getSeed(), parsed.getSignature());
+
+    // Use the parsed request to obtain the key name and type we should use to
+    // validate the signature
+    Path path = Paths.get(signature.getPublicKey().getPath());
+    Path directory = path.getParent();
+    String publicKeyPath = String.join("/", directory.toString(), parsed.getKeyid());
+
+    SignType alg;
+    try {
+      alg = SignType.fromString(parsed.getAlgorithm());
+    } catch (EnumConstantNotPresentException e) {
+      throw new AnnotatorException("Invalid key type " + parsed.getAlgorithm());
+    }
+    KeyInfo publicKey = new KeyInfo(publicKeyPath, alg);
+    signature = new SignatureInfo(publicKey, signature.getPrivateKey());
+
+    Boolean isSatisfied = verifySignature(signature.getPublicKey(), signable);
+
+    final Annotation annotation = new Annotation(
+        key,
+        hash,
+        host,
+        kind,
+        null,
+        isSatisfied,
+        Instant.now());
+
+    final String annotationSignature = super.signAnnotation(signature.getPrivateKey(), annotation);
+    annotation.setSignature(annotationSignature);
+    return annotation;
+  }
+
+  /**
+   * Responsible for verifying the signature, returns true if the verification
+   * passed, false otherwise.
+   * 
+   * @param key      The public key used to verify the signature
+   * @param signable Contains the data (seed) and signature
+   * @return True if signature valid, false otherwise
+   * @throws AnnotatorException
+   */
+  private Boolean verifySignature(KeyInfo key, Signable signable) throws AnnotatorException {
```

Comment:
This method is identical to the `verifySignature` method in the `PkiAnnotator` [here](https://github.com/mohanadismail/alvarium-sdk-java/blob/890f3fdc0abc608b806107ad1b0dca249c1c4b1c/src/main/java/com/alvarium/annotators/PkiAnnotator.java). Instead of copy pasting the method, I suggest doing one of the following to make the code as DRY as possible:

- either to create an `AbstractPkiAnnotator` , add the `verfiySignature` method to it and let the `Pki` annotator concrete classes extend from it.
- or simply add the `verifySignature` method to the [`AbstractAnnotator` class](https://github.com/mohanadismail/alvarium-sdk-java/blob/890f3fdc0abc608b806107ad1b0dca249c1c4b1c/src/main/java/com/alvarium/annotators/AbstractAnnotator.java) if it is going to be used by multiple annotators.